### DISCUSSION TITLE
VSCode Metals test discovery omits private classes

### DISCUSSION THREAD

**Discussion by LeUser111** [scalameta/metals#4016] [General]
Hi there :)

I'm currently evaluating VSCode with Metals as an alternative to Idea and I've just found an interesting difference in test discovery:

We are using ScalaTest and to keep the classes clean and separate, we are declaring all test suites/classes as `private final` by default. In Idea and maven, those tests are discovered and run without any issues. In VSCode with Metals however, the private test suites don't show up at all.

Now I'm not sure if that is a bug or somehow intended behavior. That's why I'm starting a discussion rather than opening an issue ;) 

Any insight would be appreciated!

Here are two simple examples:

This one is discovered by Maven, Idea and Metals:
```
package ...

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

final class PublicTest extends AnyFunSuite with Matchers {
  
    test("should be found") {
        true shouldBe true
    }

}
```

This one is only being discovered by Idea and Maven:
```
package ...

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

private final class PrivateTest extends AnyFunSuite with Matchers {
  
    test("should be found as well") {
        true shouldBe true
    }

}
```
As soon as I change the visibility modifier of `PrivateTest`, it shows up in the TestExplorer.

**Comment by Arthurm1** [comment]
I think this ties in with https://github.com/scalacenter/bloop/issues/996

I think Bloop and SBT deliberately exclude non-public classes from test discovery.

I understood at the time this was for performance reasons.

**Reply by LeUser111** [reply]
Thank you for digging that up! Sounds like a very likely match.

I guess I'll post a comment there. :)

Edit: I've commented on the tied issue and linked this discussion :)
